v3/unit_context.go

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package work
+
+import "go.uber.org/zap"
+
+// UnitContext represents context about a work unit passed through to work unit actions.
+type UnitContext struct {
```

Comment:
determine if we need this - could just pass an instance of `work.Unit` into `work.UnitAction`.

Reply:
after more thought, i shouldn't pass in `work.Unit`. having a separate abstraction allows for a data that may not be exported in `work.Unit` to be interacted with inside the actions.